### ISSUE TITLE
Having a ? at the start of the URL of socket requests throws TypeError

### ISSUE THREAD

**Issue by frwickst** [balderdashy/sails#2925]
**Using Sails 0.11**

**I tried using Socket.IO and connecting to Sails.js like this:**

socket.emit('get', {
  method: 'get',
  url: '?iWillCrash'
});

**This will throw the following error on the server**:
TypeError: Cannot call method 'indexOf' of null
    at Sails.session [as _privateSessionMiddleware] (<INTERNAL_PATH>/node_modules/sails/node_modules/express/node_modules/connect/node_modules/express-session/index.js:142:27)
    at loadSession (<INTERNAL_PATH>/node_modules/sails/lib/router/index.js:456:23)
    at <INTERNAL_PATH>/node_modules/sails/lib/router/index.js:199:7
    at Sails.cookieParser [as _privateCpMware] (<INTERNAL_PATH>/node_modules/sails/node_modules/express/node_modules/connect/node_modules/cookie-parser/index.js:48:5)
    at parseCookies (<INTERNAL_PATH>/node_modules/sails/lib/router/index.js:433:23)
    at <INTERNAL_PATH>/node_modules/sails/lib/router/index.js:190:5
    at qsParser (<INTERNAL_PATH>/node_modules/sails/lib/router/index.js:381:3)
    at Router.route (<INTERNAL_PATH>/node_modules/sails/lib/router/index.js:184:3)
    at Router.bound (<INTERNAL_PATH>/node_modules/sails/node_modules/lodash/dist/lodash.js:729:21)
	at receiveIncomingSailsIOMsg (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/lib/receive-incoming-sails-io-msg.js:225:16)
    at Socket.<anonymous> (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/lib/on-connect.js:84:9)
    at Socket.emit (events.js:95:17)
    at Socket.onevent (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/node_modules/socket.io/lib/socket.js:329:8)
    at Socket.onpacket (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/node_modules/socket.io/lib/socket.js:290:12)
    at Client.ondecoded (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/node_modules/socket.io/lib/client.js:193:14)
    at Decoder.Emitter.emit (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/node_modules/socket.io/node_modules/socket.io-parser/node_modules/component-emitter/index.js:134:20)
    at Decoder.add (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/node_modules/socket.io/node_modules/socket.io-parser/index.js:247:12)
    at Client.ondata (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/node_modules/socket.io/lib/client.js:175:18)
    at Socket.emit (events.js:95:17)
    at Socket.onPacket (<INTERNAL_PATH>/node_modules/sails/node_modules/sails-hook-sockets/node_modules/socket.io/node_modules/engine.io/lib/socket.js:99:14)
   ....

**Comment by sailsbot** [comment]
Thanks for posting, @frwickst.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
